synthesizer/src/vm/finalize.rs

```diff
@@ -2118,6 +2140,25 @@ finalize compute:
             assert!(expected_committee.contains(entry));
         }
 
+        // Get and check the commission mapping.
+        let actual_commission = vm.finalize_store().get_mapping_confirmed(program_id, commission_mapping_name).unwrap();
+        let expected_commission = validators
+            .iter()
+            .map(|(private_key, _)| {
+                (
+                    Plaintext::from_str(&Address::try_from(private_key).unwrap().to_string()).unwrap(),
+                    Value::from_str("0u8").unwrap(),
+                )
+            })
+            .collect_vec();
+        // Note that `actual_commission` and `expected_commission` are vectors and not necessarily in the same order.
+        // By checking that the lengths of the vector are equal and that all entries in `actual_commission` are in `expected_commission`,
+        // we can ensure that the two vectors contain the same data.
+        assert_eq!(actual_commission.len(), expected_commission.len());
+        for entry in actual_commission.iter() {
+            assert!(expected_commission.contains(entry));
+        }
```

Comment:
while this _shouldn't_ be a big deal performance-wise, it would probably be better to instead sort the vectors in advance and compare them in a single pass using `zip`